### ISSUE TITLE
Crash: Settings schema 'io.elementary.desktop.wingpanel.applications-menu' does not contain a key named 'rows'

### ISSUE THREAD

**Issue by mateHD** [elementary-tweaks/elementary-tweaks#115]
Hi guys!
I can't open Tweaks.

```
(io.elementary.switchboard:18947): Gtk-CRITICAL **: 01:04:25.043: gtk_combo_box_set_model: assertion 'GTK_IS_COMBO_BOX (combo_box)' failed
Max volume: 100

(io.elementary.switchboard:18947): GLib-GIO-ERROR **: 01:04:25.089: Settings schema 'io.elementary.desktop.wingpanel.applications-menu' does not contain a key named 'rows'
```

Could you please help me?
Thanks

**Comment by ryonakano** [comment]
I can't reproduce this issue on the latest elementary OS Juno. Are you using elementary OS Juno or Pantheon on another distro? Do you have Applications Menu installed?

**Comment by mateHD** [comment]
I'm using the latest eOS Juno with stock applications menu, wingpanel. 
It worked perfectly 2-3 months ago, when I reinstalled the OS.

**Comment by mateHD** [comment]
Can I help somehow? Do you need any other information? logs?

**Comment by ryonakano** [comment]
Hmm…I wonder why the key `rows` does not exist in your setup.

Could you check running the following command (or reboot after that if needed) fixes the issue?

```
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/
```

**Comment by mateHD** [comment]
Unfortunately it didn't help. Same error after the command and reboot.

**Comment by ryonakano** [comment]
What about reinstalling Applications Menu? :thinking:

```
sudo apt install --reinstall slingshot-launcher
```

**Comment by mateHD** [comment]
Reinstall is not helping.
my `io.elementary.desktop.wingpanel.applications-menu.gschema.xml` (if it matters)

```
<schemalist>
        <schema path="/io/elementary/desktop/wingpanel/applications-menu/" id="io.elementary.desktop.wingpanel.applications-menu">
        <key name="use-category" type="b">
            <default>false</default>
            <summary>Show the category view by default</summary>
            <description>Save the current view of the launcher.</description>
        </key>
        </schema>
</schemalist>
```
slingshot version: `slingshot-launcher (2.4.4+r1195+pkg80~daily~ubuntu5.0.1)`
wingpanel version: `wingpanel (2.2.6+r420+pkg44~daily~ubuntu5.0.1)`
elementary-tweaks version: `elementary-tweaks  (1.0-0~249~ubuntu5.0.1)`

**Comment by ryonakano** [comment]
OK, I found the reason of this issue and why I couldn't reproduce it. You seem to add the daily PPA of elementary at some point and in the latest version of **daily** (not Juno), [the size configuration option seems to be removed](https://github.com/elementary/applications-menu/commit/c0471d24f38ba3192adb50b5c5d76fe1c32635eb#diff-1483c2984b40a030cae1b018341d060a) and at the same time the keys `rows` and `columns` were removed. This is the reason why those keys aren't found and the Plug crashes.

**Comment by ryonakano** [comment]
Since the removal of the size configuration option was done in Applications Menu itself, the realistic solution to fix this issue is to remove this option from elementary tweaks too. To revive the option in Applications Menu would be mostly impossible.

**Comment by mateHD** [comment]
Oh, okay. i'm sorry for taking your time, I thought its still Juno.
I hope there will be an update, I'm waiting.

**Comment by ryonakano** [comment]
I've made a PR that fixes this issue, but I'm not sure it is merged. Many PRs are pended.

**Comment by mateHD** [comment]
Hi ryonakano,
I built the Tweaks with your changes, but sadly It is still crashing.
```
Max volume: 100

(io.elementary.switchboard:17973): GLib-GIO-ERROR **: 09:44:45.534: Settings schema 'io.elementary.desktop.wingpanel.applications-menu' does not contain a key named 'rows'

```
Good news: first error (`(io.elementary.switchboard:18947): Gtk-CRITICAL **: 01:04:25.043: gtk_combo_box_set_model: assertion 'GTK_IS_COMBO_BOX (combo_box)' failed`) disappeared.

**Comment by ryonakano** [comment]
@mateHD You don't have to close the issue. We can close it when the PR gets merged.

**Comment by leandrocodes** [comment]
Waiting for the merge... same error here. Woked perfectly until yesterday

**Comment by Philip-Scott** [comment]
Thank you for the reminder, I just merged the PR and hopefully by tomorrow it will be deployed in the PPA :) 

Thank you for your fix @ryonakano, and sorry for the slow reviews.... I've been super busy lately with the new job and everything :(

**Comment by leandrocodes** [comment]
This is a great news!

**Comment by jimmybe123** [comment]
linux newbie here, so to fix this issue i just have to wait for the update to come out then- sudo apt update and sudo apt upgrade in terminal?
thanks.

**Comment by ryonakano** [comment]
@jimmybe123 Yes.

**Comment by manssorr** [comment]
linux newbie here too, What that means?

Err:10 http://ppa.launchpad.net/mpstark/elementary-tweaks-daily/ubuntu xenial Release         
  404  Not Found [IP: 91.189.95.83 80]
Get:11 http://eg.archive.ubuntu.com/ubuntu bionic-backports InRelease [74.6 kB]               
Reading package lists... Done        
E: The repository 'http://ppa.launchpad.net/mpstark/elementary-tweaks-daily/ubuntu xenial Release' does not have a Release file.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.

**Comment by ryonakano** [comment]
@manssorr I guess you've added an PPA provided by another person.
Try:

```
sudo add-apt-repository --remove -y ppa:mpstark/elementary-tweaks-daily
sudo add-apt-repository -y ppa:philip.scott/elementary-tweaks
sudo apt -y install elementary-tweaks
```

**Comment by alissontenorio** [comment]
If anyone is interested: While ppa is not updated, as a workround I used information that @ryonakano provided and make Tweaks work again:

Insert on /usr/share/glib-2.0/schemas/io.elementary.desktop.wingpanel.applications-menu.gschema.xml :

```
<key name="columns" type="i">
            <default>5</default>
            <summary>The default number of columns</summary>
            <description>This value needs to be greater than 3 to be effective.</description>
</key>
<key name="rows" type="i">
            <default>3</default>
            <summary>The default number of rows</summary>
            <description>This value needs to be greater than 1 to be effective.</description>
</key>

```

After that do recompile the schemas using:
sudo glib-compile-schemas /usr/share/glib-2.0/schemas/

This should fix the issue for now. Remember to remove what you inserted in io.elementary.desktop.wingpanel.applications-menu.gschema.xml after the ppa is updated with the real solution.

**Comment by wprauchholz** [comment]
Would somebody please tell me how to install it directly via git?

Thank you

**Comment by ryonakano** [comment]
@wprauchholz It's written in [the README](https://github.com/elementary-tweaks/elementary-tweaks#how-to-build), but I don't recommend installing with git if you're not accustomed to this kind of things. When Philip-Scott publishes the fix you have to remove the manual installation with git and then *sudo apt dist-upgrade*. I guess the way alissontenorio described would be more safe and easy.

**Comment by alissontenorio** [comment]
PPA is updated

**Comment by tudorels** [comment]
In new build Hera, is not working. The launcher remain with 5 columns and 3 rows.

**Comment by ryonakano** [comment]
Hi @tudorels, the option to change the number of columns and rows [was removed](https://github.com/elementary/applications-menu/commit/c0471d24f38ba3192adb50b5c5d76fe1c32635eb#diff-1483c2984b40a030cae1b018341d060a) from the launcher. Because it would be hard to revive that option in the launcher, we decided to remove settings also from elementary-tweaks (otherwise we face a crash when opening elementary-tweaks). So we can no longer change the launcher size anymore, unfortunately.
